.github/workflows/rust-checks.yml

```diff
@@ -0,0 +1,56 @@
+name: Rust Checks
```

Comment:
I removed `.github/workflows/test-sp1-docker.yml` for this new file.
The idea is now that since running tests also uses Docker for compiling guest programs, then the image building process is already indirectly tested.

And actually, in our previous workflows we didn't run any tests. Nor did some basic lint checks, so we have other benefits here.

I think as we migrate more zkVMs to use docker, we can simply keep editing this file to consider more and more `ere-*` crates.